src/main/resources/META-INF/rewrite/spring-data3-upgrade.yml

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright 2022 the original author or authors.
+# <p>
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://www.apache.org/licenses/LICENSE-2.0
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+---
+########################################################################################################################
+# Spring Data 3.0
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.spring.boot3.data.UpgradeSpringData_3_0
+displayName: Upgrade to Spring Data 3.0
+description: 'Upgrade to Spring Data to 3.0 from any prior version.'
+recipeList:
+  - org.openrewrite.java.spring.boot3.data.SwitchToJakarta
```

Comment:
rewrite-migrate-java has a JavaxToJakarta recipe which can be used instead
https://github.com/openrewrite/rewrite-migrate-java/blob/main/src/main/resources/META-INF/rewrite/javax-to-jakarta.yml

Created an issue for adding a new recipe to support the xml-binding 4.0.0 api.
https://github.com/openrewrite/rewrite-migrate-java/issues/110